google-cloud-firestore/src/main/java/com/google/cloud/firestore/BulkWriter.java

```diff
@@ -59,13 +61,15 @@
      * @param error The error object from the failed BulkWriter operation attempt.
      * @return Whether to retry the operation or not.
      */
-    boolean shouldRetryListener(BulkWriterError error);
+    boolean shouldRetryError(BulkWriterError error);
```

Comment:
I am thinking more and more that this should be `onError`. Otherwise, users are going to ask us how they can get the error result and why we don't expose it in  `WriteResultCallback`. Note also that the term `result` is a bit ambiguous as it doesn't define whether "error" is a "result".

We should maybe also ask for a third opinion of whether the classes are "callbacks".  On Android, we seem to call the outermost class "Listener". The method inside is prefixed with "on" (see EventListener).

Reply:
Changed to `onError`. As for WriteResult, we discussed naming for the node implementation and decided on `WriteResult` over `WriteSuccess` since a `WriteResult` is returned from the operation.

Would it be too sadistic to ask Ben to take a look at this PR for Java code style/convention aftewards?

Reply:
We should definitely ask Ben :)